### PR TITLE
[PSM Interop] Loosen affinity test assertions

### PR DESCRIPTION
**Pull request by gnossen**
As part of the dualstack backend designs, subchannels will be created lazily. Therefore, instead of asserting that there is 1 READY subchannel and `n - 1` IDLE subchannels, we just assert that there is 1 READY subchannel.

ref b/290264214

### CODE REVIEW

File: tools/run_tests/xds_k8s_test_driver/tests/affinity_test.py [hunk 1]
```diff
@@ -143,20 +143,7 @@ def test_affinity(self) -> None:  # pylint: disable=too-many-statements
                     f" {len(ready_channels)} instead."
                 ),
             )
-            # The rest should be IDLE.
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/tests/affinity_test.py:146]: Might be helpful to add a note at line 133 that the rest may be in any state.

**Review by sergiitk — APPROVED** [review]
LGTM once tests pass:

- [x] [grpc/core/master/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/9a11b2c8-a58d-487c-8cda-4615b35f3d00)
- [ ] [grpc/core/master/linux/grpc_xds_url_map](https://source.cloud.google.com/results/invocations/6e7392a3-367b-4f77-aae2-b150952a6aff) (failed)

**Review by sergiitk — CHANGES_REQUESTED** [review]
We need to update urlmap test as well:

https://source.cloud.google.com/results/invocations/6e7392a3-367b-4f77-aae2-b150952a6aff/targets/grpc%2Fcore%2Fmaster%2Flinux%2Fgrpc_xds_url_map%2Furl_map/tests

```py
Traceback (most recent call last):
  File "/tmpfs/tmp/tmp.ZGpp6EB8Zy/grpc/tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py", line 499, in test_rpc_distribution
    self.rpc_distribution_validate(self.test_client)
  File "/tmpfs/tmp/tmp.ZGpp6EB8Zy/grpc/tools/run_tests/xds_k8s_test_driver/tests/url_map/affinity_test.py", line 129, in rpc_distribution_validate
    self.assertLen(
AssertionError: [] has length of 0, expected 2.
```

**Comment by gnossen** [comment]
Good catch. Turns out the `url_map` version has several such assertions. Fixed.

**Comment by gnossen** [comment]
Looks green.

**Review by sergiitk — APPROVED** [review]

**Pull request merged**